src/models/Artifact.ts

```diff
@@ -1,34 +1,70 @@
-export type ArtifactData = Record<string, unknown>
+export enum ArtifactType {
+  Result = 'result',
+  Markdown = 'markdown',
+  Table = 'table',
+  Unknown = 'unknown'
+}
```

Comment:
Can we use a string union rather than an enum? We've avoided using enums because they have weird side effects.

Reply:
can you expand on the weird side effects piece? We have type guards so this isn't strictly necessary but it's certainly a cleaner expectation

Reply:
Sadly a video is the best explanation I can give. https://www.youtube.com/watch?v=jjMbPt_H3RQ

We currently intentionally haven't used enums in any of our projects and unless we have a good reason to do so I would rather be consistent and use an array const with a type. 

I also personally prefer string unions because with an enum you have to import the enum everywhere you want to use one of the values. A string union gives you all the type safety without the additional import.

Reply:
Nice that's helpful - I'll make that update